src/test/resources/configurations/configuration-1-redeploy.json

```diff
@@ -0,0 +1,44 @@
+{
+  "components": {
+    "MyThermostatComponent": {
+      "version": "1.0.0",
+      "configurationUpdate": {
+        "MERGE": {
+          "units": "C",
+          "rooms": {
+            "Kitchen": {
+              "temperature": 25,
+              "fan": "on"
+            },
+            "Office": {
+              "temperature": 22,
+              "fan": "auto"
+            },
+            "Conference Room": {
+              "temperature": 20,
+              "fan": "auto"
+            }
+          }
+        },
+        "RESET": [
+          "/rooms/Kitchen",
+          "/rooms/Office",
+          "/rooms/Conference Room"
+        ]
+      },
+      "runWith": {
+        "posixUser": "user:group"
+      }
+    }
+  },
+  "failureHandlingPolicy": "DO_NOTHING",
+  "componentUpdatePolicy": {
+    "timeout": "5000",
+    "action": "SKIP_NOTIFY_COMPONENTS"
+  },
+  "configurationValidationPolicy": {
+    "timeout": "30000"
+  },
+  "baseTargetArn": "arn:aws:iot:us-east-1:123456789012:thinggroup/parentGroup",
+  "rootTargetArn": "arn:aws:iot:us-east-1:123456789012:thinggroup/parentGroup"
```

Comment:
what's the difference between base and root? That's not very clear at all.

Reply:
The rootTargetArn would represent the targetArn of the fleet, where as baseTargetArn is the immediate parent. In multi-level sub-group deployment these would be different. Both of these values would be added to deployment configuration by cloud.

Reply:
OK, thank you for explaining. I think these need to be renamed.

I like `onBehalfOf`: this is the group that will be used when doing the dependency resolution, we're pretending that the deployment comes from this group.

For the other one, maybe just `parentTargetArn`?

Reply:
I think what Michael means is, and I also want to see, if naming can be better

Reply:
The terminology for this feature is yet to be agreed upon. I can use the proposed names and update the PR.